testing/base/file_helpers.h

```diff
@@ -15,6 +15,23 @@ namespace Carbon::Testing {
 // Reads a file to string.
 auto ReadFile(std::filesystem::path path) -> ErrorOr<std::string>;
 
+// Writes a test file to disk and returns an error or the full path to the file.
+//
+// Note that this expects to be run from within a GoogleTest test, and relies on
+// global state of GoogleTest.
+//
+// This locates a suitable temporary directory for the test, creates a file with
+// the requested name in that directory, and writes the provided content to that
+// file. The full path to the written file is returned.
+//
+// Where possible, this will use a Bazel-provided test temporary directory.
+// However, if unavailable, it falls back to a system temporary directory. This
+// helps tests be runnable outside of Bazel, for example under a debugger. It
+// also works to create test file names that are unlikely to conflict with other
+// tests when run.
+auto WriteTestFile(llvm::StringRef name, llvm::StringRef contents)
```

Comment:
Just to note, if you touch this again, you might call it `WriteFile` to match `ReadFile` (it's already in the `Testing` namespace)